### PR TITLE
[#1323] - Add circleCI

### LINKED ISSUE #1323: Trial for CircleCI
I think we should consider to move to CircleCI as replacement for TravisCI:
* Cleaner UI
* Faster to build

I think we should compare them for a week and then decide... I don't expect CircleCI & TravisCI has a massive difference in the behavior because we use docker.

### PR DESCRIPTION
**Pull request by Saphyel**
**Type**: patch

Fixes #1323

### CODE REVIEW

**Comment by Saphyel** [comment]
Seems like there is an issue with behat when you export to junit.
I'm not sure about the behavior if fail a test and you ignore the output of behat because the summary will report failing tests

**Comment by Saphyel** [comment]
@pvhee now both of them failed (as epxected)

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+          name: Set up the project
+          command: |
+            docker-compose exec web composer campaign:build || true
+            docker-compose exec web chmod ugo+rw web/sites/default -R
+      - run:
+          name: Run the test
+          command: |
+            docker-compose exec web sh -c 'cd web && drush pm-uninstall cookieconsent toolbar -y'
+            docker-compose exec web vendor/bin/behat -n
+            docker-compose exec web vendor/bin/behat -ns rest
```
Comment (pvhee) [inline .circleci/config.yml]: @Saphyel why are you not using https://github.com/comicrelief/campaign/blob/master/ci/travis/run-tests.sh ? like you do for the dberrors script? Maybe we should refactor that and just put these in the `ci/` directory so both Travis / Circle can use these?

Comment (Saphyel) [reply .circleci/config.yml]: @pvhee because I was testing the junit thing, and travis is not able to do anything with the reports.

**Comment by pvhee** [comment]
@Saphyel can you also add a CircleCI batch to the README?

**Comment by Saphyel** [comment]
do you want to keep the same folder structure? I think maybe we can leave it as it is until we have to kick one of the CI

**Comment by pvhee** [comment]
We can leave the folder structure

**Comment by Saphyel** [comment]
Come on travis...

**Pull request merged**